### DISCUSSION TITLE
Support for Manually Customizing Models When API Providers Don't Support /v1/models Endpoint

### DISCUSSION THREAD

**Discussion by zhiqing0205** [open-webui/open-webui#4617] [General]
**Is your feature request related to a problem? Please describe.**
I'm frustrated because some API providers do not support the `/v1/models` endpoint to retrieve available models, but they do support the `/v1/chat/completions` endpoint for normal chat interactions. This limitation prevents the use of certain API providers with OpenWebUI, as it currently relies on the `/v1/models` endpoint to detect available models.

**Describe the solution you'd like**
I would like to request the ability to manually customize or input a model in the top left corner of the interface. This feature would allow users to specify the model they want to use when the API provider does not support the `/v1/models` endpoint, enabling broader compatibility with different API providers.

**Describe alternatives you've considered**
An alternative could be implementing a fallback mechanism where, if the `/v1/models` endpoint is not supported by the API provider, users could still input a model manually or select from a predefined list of commonly used models.

This enhancement would greatly improve flexibility and compatibility with a wider range of API providers.

**Comment by moblangeois** [comment]
In addition, I'd like to point out that it is the case with perplexity API.

**Comment by tjbck** [comment]
Should be doable with Functions, I'd love to see contributions here!

Docs: https://docs.openwebui.com/tutorial/functions
Examples: https://openwebui.com/functions
